lib/Basics/error.cpp

```diff
@@ -49,7 +58,19 @@ std::string_view TRI_last_error() {
   ErrorCode err = LastError._number;
 
   if (err == TRI_ERROR_SYS_ERROR) {
+#if defined(_WIN32)
+    if (LastError._sys == 0) {
+      return {};
+    }
+
+    static CHAR messageBuffer[sizeof(TCHAR) * 1024] = {0};
```

Comment:
It looks like using a static buffer is not thread-safe, in case `TRI_last_error()` is called from multiple threads in parallel, and they all write to the same buffer.
I guess we should remove the `static` declaration here.

Reply:
Good catch! I changed the parameters of `FormatMessageA` such that it allocates the buffer itself. I had to change the return type of `TRI_last_error` to `std::string` in order to support the Windows counterpart. As we discussed, it should not impact performance, as it is used in case an error occurs.